pydantic_ai_slim/pydantic_ai/models/openai.py

```
@@ -4631,3 +4705,51 @@ def _map_mcp_call(
+) -> FilePart | None:
+    """Download a file the code interpreter wrote to its container, or `None` if it can no longer be retrieved.
+
+    Containers are short-lived, so a download can fail (expired container, rate limit, transient error). In that case
+    we skip the file with a warning rather than discarding the otherwise-successful response.
+    """
+    try:
+        file = await client.containers.files.content.retrieve(
+            file_id=citation.file_id, container_id=citation.container_id
+        )
+    except (APIStatusError, APIConnectionError) as e:
+        warnings.warn(
```

Ruff (0.15.20)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)